src/compiler/checker.ts

```diff
@@ -23007,13 +23007,20 @@ namespace ts {
 
                 case SyntaxKind.NumericLiteral:
                     // index access
-                    if (node.parent.kind === SyntaxKind.ElementAccessExpression && (<ElementAccessExpression>node.parent).argumentExpression === node) {
-                        const objectType = getTypeOfExpression((<ElementAccessExpression>node.parent).expression);
-                        return getPropertyOfType(objectType, (<NumericLiteral>node).text as __String);
-                    }
-                    break;
+                    const objectType = (() => {

```

Comment:
the IIFE seems like overkill for what could be `let objectType; switch ...` with some assignments to `objectType` inside. Or does it introduce some names that conflict with the outside scope?